core/src/test/java/org/apache/iceberg/puffin/TestFileMetadataParser.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.puffin;
+
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestFileMetadataParser {
```

Comment:
maybe add a few tests with invalid/missing json fields?

Reply:
good idea.

this let me realize there is a problem with trailing contentn

i.e. this should fail:
```java
FileMetadataParser.fromJson("{\"blobs\": []} {}")
```

but it does not.

I think i am following the established JSON parsing patterns, so it probably is a pre-existing problem

Reply:
> I think i am following the established JSON parsing patterns, so it probably is a pre-existing problem

https://github.com/apache/iceberg/pull/5048